l2-contracts/contracts/bridge/L2SharedBridge.sol

```diff
@@ -37,147 +38,155 @@ contract L2SharedBridge is IL2SharedBridge, Initializable {
-        address _aliasedOwner
+        IL2StandardDeployer _standardDeployer
     ) external reinitializer(2) {
         require(_l1SharedBridge != address(0), "bf");
-        require(_l2TokenProxyBytecodeHash != bytes32(0), "df");
-        require(_aliasedOwner != address(0), "sf");
+        require(address(_standardDeployer) != address(0), "cf");
 
         l1SharedBridge = _l1SharedBridge;
-
-        if (block.chainid != ERA_CHAIN_ID) {
-            address l2StandardToken = address(new L2StandardERC20{salt: bytes32(0)}());
-            l2TokenBeacon = new UpgradeableBeacon{salt: bytes32(0)}(l2StandardToken);
-            l2TokenProxyBytecodeHash = _l2TokenProxyBytecodeHash;
-            l2TokenBeacon.transferOwnership(_aliasedOwner);
-        } else {
+        standardDeployer = _standardDeployer;
+        if (block.chainid == ERA_CHAIN_ID) {
             require(_l1Bridge != address(0), "bf2");
             l1Bridge = _l1Bridge;
-            // l2StandardToken and l2TokenBeacon are already deployed on ERA, and stored in the proxy
         }
     }
 
     /// @notice Finalize the deposit and mint funds
     /// @param _l1Sender The account address that initiated the deposit on L1
-    /// @param _l2Receiver The account address that would receive minted ether
+    // / @param _l2Receiver The account address that would receive minted ether
     /// @param _l1Token The address of the token that was locked on the L1
-    /// @param _amount Total amount of tokens deposited from L1
+    // / @param _amount Total amount of tokens deposited from L1
     /// @param _data The additional data that user can pass with the deposit
-    function finalizeDeposit(
-        address _l1Sender,
-        address _l2Receiver,
-        address _l1Token,
-        uint256 _amount,
-        bytes calldata _data
-    ) external override {
+    function finalizeDeposit(bytes32 _assetInfo, bytes memory _data) public override {
         // Only the L1 bridge counterpart can initiate and finalize the deposit.
+
         require(
             AddressAliasHelper.undoL1ToL2Alias(msg.sender) == l1Bridge ||
                 AddressAliasHelper.undoL1ToL2Alias(msg.sender) == l1SharedBridge,
             "mq"
         );
 
-        address expectedL2Token = l2TokenAddress(_l1Token);
-        address currentL1Token = l1TokenAddress[expectedL2Token];
-        if (currentL1Token == address(0)) {
-            address deployedToken = _deployL2Token(_l1Token, _data);
-            require(deployedToken == expectedL2Token, "mt");
-            l1TokenAddress[expectedL2Token] = _l1Token;
+        address asset = assetAddress[_assetInfo];
+        if (asset != address(0)) {
+            IL2StandardAsset(asset).bridgeMint(L1_CHAIN_ID, _assetInfo, _data);
         } else {
-            require(currentL1Token == _l1Token, "gg"); // Double check that the expected value equal to real one
+            IL2StandardAsset(standardDeployer).bridgeMint(L1_CHAIN_ID, _assetInfo, _data);
```

Comment:
Just so we dont forget, it is better if the L1NTV could somehow set the correct assetAddress, so that the logic is uniform for NTV and the others